autoupgrade.php

```diff
@@ -149,9 +149,58 @@ public function uninstall()
         // Remove the 1-click upgrade working directory
         self::_removeDirectory(_PS_ADMIN_DIR_ . DIRECTORY_SEPARATOR . 'autoupgrade');
 
+        Configuration::deleteByName('PS_AUTOUP_IGNORE_REQS');
+        Configuration::deleteByName('PS_AUTOUP_IGNORE_PHP_UPGRADE');
+
         return parent::uninstall();
     }
 
+    private function registerHookAndSetToTop($hookName)
+    {
+        if ($this->registerHook($hookName)) {
+            // Update module position in Dashboard
+            $query = 'SELECT id_hook FROM ' . _DB_PREFIX_ . "hook WHERE name = '" . pSQL($hookName) . "'";
+            $result = Db::getInstance()->ExecuteS($query);
```

Comment:
use `getRow` instead? or better `getIdByName`?

Reply:
What a good idea. The method is now really simpler